App crash after process restart

We are on 2.alpha-1 version.

We have a couple of dev and production cases when we restart the app with the help of the Phoenix rebirth library.

We noticed that app doesn't start and looking to the logcat we found it is crashing now:
```

Process: mycomp:phoenix, PID: 12928
    java.lang.ExceptionInInitializerError
        at mycomp.libs.leakcanary.LoveLeakCanary.install(LoveLeakCanary.kt:21)
        at mycomp.libs.leakcanary.startup.RealLeakCanaryInitializer.initialize(RealLeakCanaryInitializer.kt:10)
        at mycomp.android.common.startup.ApplicationStartupTasks.installLeakCanary(ApplicationStartupTasks.kt:95)
        at mycomp.android.common.startup.ApplicationStartupTasks.initFrameworks(ApplicationStartupTasks.kt:76)
        at mycomp.android.common.startup.ApplicationStartupTasks.execute(ApplicationStartupTasks.kt:42)
        at mycomp.android.LoveApplication.onCreate(LoveApplication.kt:79)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property application has not been initialized
        at leakcanary.internal.InternalLeakCanary.getApplication(InternalLeakCanary.kt:34)
        at leakcanary.LeakCanary$Config.<init>(LeakCanary.kt:13)
        at leakcanary.LeakCanary.<clinit>(LeakCanary.kt:22)

```

So looks like the content provider mechanism doesn't work with this case.

--- a/leakcanary-android-core/src/main/java/leakcanary/LeakCanary.kt
+++ b/leakcanary-android-core/src/main/java/leakcanary/LeakCanary.kt
@@ -50,7 +50,7 @@ object LeakCanary {
   )
 
   @Volatile
-  var config: Config = Config()
+  var config: Config = if (LeakSentry.isInstalled) Config() else InternalLeakCanary.noInstallConfig
 
   /** [Intent] that can be used to programmatically launch the leak display activity. */
   val leakDisplayActivityIntent

--- a/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
+++ b/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
@@ -14,6 +14,7 @@ import com.squareup.leakcanary.core.R
 import leakcanary.CanaryLog
 import leakcanary.GcTrigger
 import leakcanary.LeakCanary
+import leakcanary.LeakCanary.Config
 import leakcanary.LeakSentry
 import leakcanary.internal.activity.LeakActivity
 
@@ -34,6 +35,12 @@ internal object InternalLeakCanary : LeakSentryListener {
 
   val leakDisplayActivityIntent: Intent
     get() = LeakActivity.createIntent(application)
+
+  val noInstallConfig: Config
+    get() = Config(
+        dumpHeap = false, exclusionsFactory = { emptyList() }, leakInspectors = emptyList(),
+        labelers = emptyList()
+    )
 
   override fun onLeakSentryInstalled(application: Application) {
     this.application = application

--- a/leakcanary-leaksentry/src/main/java/leakcanary/LeakSentry.kt
+++ b/leakcanary-leaksentry/src/main/java/leakcanary/LeakSentry.kt
@@ -15,16 +15,21 @@ object LeakSentry {
   )
 
   @Volatile
-  var config: Config = Config()
+  var config: Config = if (isInstalled) Config() else Config(enabled = false)
 
   val refWatcher
     get() = InternalLeakSentry.refWatcher
 
+  /** @see [manualInstall] */
+  val isInstalled
+    get() = InternalLeakSentry.isInstalled
+
   /**
-   * [LeakSentry] is automatically installed on process start by
-   * [leaksentry.internal.LeakSentryInstaller] which is registered in the AndroidManifest.xml of
-   * your app. If you disabled [leaksentry.internal.LeakSentryInstaller] then you can call this
-   * method to install [LeakSentry].
+   * [LeakSentry] is automatically installed on main process start by
+   * [leakcanary.internal.LeakSentryInstaller] which is registered in the AndroidManifest.xml of
+   * your app. If you disabled [leakcanary.internal.LeakSentryInstaller] or you need LeakSentry
+   * or LeakCanary to run outside of the main process then you can call this method to install
+   * [LeakSentry].
    */
   fun manualInstall(application: Application) = InternalLeakSentry.install(application)
 

--- a/leakcanary-leaksentry/src/main/java/leakcanary/internal/InternalLeakSentry.kt
+++ b/leakcanary-leaksentry/src/main/java/leakcanary/internal/InternalLeakSentry.kt
@@ -13,6 +13,9 @@ import java.util.concurrent.Executor
 
 internal object InternalLeakSentry {
 
+  val isInstalled
+    get() = ::application.isInitialized
+
   private val listener: LeakSentryListener
 
   val isDebuggableBuild by lazy {

